packages/blockchain-link/src/workers/blockbook/websocket.ts

```diff
@@ -433,13 +423,11 @@ export class BlockbookAPI extends TypedEmitter<BlockbookEvents> {
             clearTimeout(this.connectionTimeout);
         }
 
-        const { ws } = this;
         if (this.isConnected()) {
             this.disconnect();
         }
-        if (ws) {
-            ws.removeAllListeners();
-        }
+        this.ws?.removeAllListeners();
+        this.removeAllListeners();
```

Comment:
I'm generally against removing listeners from inside.

This way, we're not able to catch `disconnected` event from outside because the listener is usually removed right before that. If I remember correctly, that was the reason why I separated `onClose` (could be called from inside) and `dispose` (always called from outside so we explicitly declare we don't care about events anymore).

Also, if we don't remove the listeners in case of error or closing, `BlockbookAPI` instance could be reconnected and reused.

Reply:
> This way, we're not able to catch disconnected event from outside because the listener is usually removed right before that.

ok, i was just unifying things with BlockfrostApi (blockfrost/websocket.ts)
i wasn't aware that you did separate `onClose` and `dispose` - just for the future it would be nice to keep those two files consistent since they are basically the same with different messages

fixup [6462108](https://github.com/trezor/trezor-suite/pull/8718/commits/6462108d74fa6b7ab1940cbe64198cfbc6a301b1)

Reply:
Agreed, I may try to deduplicate the code of these two as a followup.